packages/frontend/src/components/ConnectivityToast.tsx

```diff
@@ -71,6 +75,10 @@ export default function ConnectivityToast() {
     [maybeNetwork]
   )
 
+  // Update ref to enable recursive calls without circular dependency
+  // eslint-disable-next-line react-hooks/refs
+  tryMaybeNetworkIfOfflineAfterXmsRef.current = tryMaybeNetworkIfOfflineAfterXms
+
```

Comment:
What's the point of all this complexity if we have to disable the linter anyway?
My suggestion below should be a little simpler, although the entire function looks like it needs to be rewritten.

```suggestion
  const tryMaybeNetworkIfOfflineAfterXms = useCallback(
    function tryMaybeNetworkIfOfflineAfterXms_(ms: number) {
      setTimeout(() => {
        // This is a hack to get the current network state by abusing the setNetworkState function.
        // Not pretty but works.
        setNetworkState(([connectivity, error]: [Connectivity, string]) => {
          if (connectivity === Connectivity.NOT_CONNECTED) {
            log.debug(
              `We are still not online after ${ms}  milli seconds, try maybeNetwork again`
            )
            maybeNetwork()
          } else if (ms < 30000) {
            tryMaybeNetworkIfOfflineAfterXms_(2 * ms)
          } else {
            log.debug(
              `We tried reconnecting with waiting for more then 30 seconds, now stop`
            )
          }

          // Keep state unchanged
          return [connectivity, error]
        })
      }, ms)
    },
    [maybeNetwork]
  )

```